src/extra/others/fragment/lv_fragment_manager.c

```diff
@@ -0,0 +1,346 @@
+{
+    lv_fragment_manager_t * instance = lv_mem_alloc(sizeof(lv_fragment_manager_t));
+    lv_memset_00(instance, sizeof(lv_fragment_manager_t));
+    instance->parent = parent;
+    return instance;
+}
+
+void lv_fragment_manager_del(lv_fragment_manager_t * manager)
+{
+    LV_ASSERT_NULL(manager);
+    lv_fragment_managed_states_t * top = manager->top;
+    while(top) {
+        LV_ASSERT(top->cls);
+        item_del_obj(top);
+        item_del_fragment(top);
+        lv_fragment_managed_states_t * prev = top->prev;
+        lv_mem_free(top);
+        top = prev;
+    }
+    lv_mem_free(manager);
+}
+
+void lv_fragment_manager_add(lv_fragment_manager_t * manager, lv_fragment_t * fragment, lv_obj_t * const * container)
+{
+    lv_fragment_managed_states_t * item = fragment_attach(manager, fragment, container);
+    item_create_obj(item, NULL);
+}
+
+void lv_fragment_manager_remove(lv_fragment_manager_t * manager, lv_fragment_t * fragment)
+{
+    LV_ASSERT(manager);
+    LV_ASSERT(fragment);
+    LV_ASSERT_NULL(fragment->managed);
+    LV_ASSERT(fragment->managed->manager == manager);
+    lv_fragment_managed_states_t * states = fragment->managed;
+    lv_fragment_managed_states_t * prev = NULL;
+    if(states->in_stack) {
+        lv_fragment_managed_states_t * next;
+        for(next = states->is_msgbox ? manager->msgbox_top : manager->top; next; next = next->prev) {
+            if(next->prev == states) {
+                break;
+            }
+        }
+        prev = states->prev;
+        if(next) {
+            next->prev = prev;
+        }
+        else if(states->is_msgbox) {
+            /*Removing top states*/
+            manager->msgbox_top = prev;
+        }
+        else {
+            manager->top = prev;
+        }
+    }
+    item_del_obj(states);
+    item_del_fragment(states);
+    lv_mem_free(states);
+    if(prev) {
+        item_create_obj(prev, NULL);
+    }
+}
+
+void lv_fragment_manager_push(lv_fragment_manager_t * manager, lv_fragment_t * fragment, lv_obj_t * const * container)
+{
+    lv_fragment_managed_states_t * top = manager->top;
+    if(top != NULL) {
+        item_del_obj(top);
+    }
+    lv_fragment_managed_states_t * states = fragment_attach(manager, fragment, container);
+    states->prev = top;
+    states->in_stack = true;
+    manager->top = states;
+    item_create_obj(states, NULL);
+}
+
+void lv_fragment_manager_pop(lv_fragment_manager_t * manager)
+{
+    lv_fragment_manager_remove(manager, lv_fragment_manager_get_top(manager));
+}
+
+void lv_fragment_manager_replace(lv_fragment_manager_t * manager, lv_fragment_t * fragment,
+                                 lv_obj_t * const * container)
+{
+    lv_fragment_t * top = lv_fragment_manager_get_top(manager);
+    if(top != NULL) {
+        lv_fragment_manager_remove(manager, top);
+    }
+    lv_fragment_manager_push(manager, fragment, container);
+}
+
+void lv_fragment_manager_recreate_obj(lv_fragment_manager_t * manager, lv_fragment_t * fragment)
```

Comment:
Isn't it redundant with `lv_fragment_recreate_obj()`?

Reply:
`lv_fragment_recreate_obj` is a shorthand to `lv_fragment_manager_recreate_obj`. How about to use inline function instead?

Reply:
* seems I shouldn't. I think it's OK to leave this as is.